rfd/0097-teleport-connect-usage-metrics.md

```diff
@@ -0,0 +1,221 @@
+differentiate events coming from multiple application instances. To do this, each event will have `distinct_id` field.
+It will be supplied with UUID generated by Connect, which will be generated once and stored in a file in the _App Data_
+directory, so it won't change between restarts.
+
+### User agreement
+
+On the start, Teleport Connect will ask user to opt in to volunteer anonymized metrics and usage-data with standard
+message "Are you OK sending anonymized usage data about Teleport Connect? This will help us to improve product".
+
+If the user refuses, no data will be collected.
+
+### How will collecting metrics support product development?
+
+In the initial version, it should help with getting answers to the following questions:
+
+#### How many unique users download and use Teleport Connect today?
+
+To answer the first part of the question, download counts from `goteleport.com/download` are needed.
+This task should be done by the marketing team.
+
+To calculate how many users use Teleport Connect on a daily basis, a metric like DAU (Daily Active Users) can be used.
+This metric can be based on a specific event, but in this case it should be calculated using any event. For example,
+user logged only once in a given day to refresh certs for a DB proxy connection - such user can be considered as active.
+
+#### What features are the most popular?
+
+Usage of each feature will be measured basing on events from the [events](#events) section. They will allow to generate
+various statistics, like the most common opened kinds of tabs or just show the usage of particular UI elements.
+
+#### Where users struggle the most?
+
+//TODO
+
+#### What platforms are the most popular?
+
+This will be measured in two ways.
+
+- Based on downloads count for each platform.
+- Based on a real usage. Connect will emit _start app_ event that contains OS field. These events will be then
+  aggregated by unique users.
+
+#### How usage grows or shrinks over time?
+
+PostHog allows to create [Trends](https://posthog.com/manual/trends) basing on DAU. It can be used to show how usage
+changes in a given period of time.
+
+## Events
+
+### `connect.start`
+
+Starting the app.
+
+Event properties:
+
+- `os`: one of `mac`/`windows`/`linux`
+- `connect_version`: string
+
+### `connect.cluster.add`
+
+Successful adding a cluster.
+
+Event properties:
+
+- `root_cluster_name`: string - name of the added cluster
+
+### `connect.cluster.login`
+
+Successful login to a cluster.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `connector_type`: string
+
+### `connect.workspace.change`
+
+Changing a workspace.
+
+Event properties:
+
+- `target_root_cluster_name`: string
+
+### `connect.leafClusterPicker.change`
+
+Changing a leaf cluster.
+
+Event properties:
+
+- `root_cluster_name`: string
+- `target_leaf_cluster_name`: string
+
+### `connect.tab.open`
+
+Opening a tab.
```

Comment:
Same here, instead of tracking tab opens, we should track on the client how many protocols were used per connect session, e.g. before exit, and send one event with counters.